render/render.js

```diff
@@ -119,13 +119,15 @@ module.exports = function($window) {
 
 		if (vnode.attrs != null) initLifecycle(vnode.attrs, vnode, hooks)
 		initLifecycle(vnode.state, vnode, hooks)
-		vnode.instance = Vnode.normalize(vnode.state.view(vnode))
+		var output = vnode.state.view(vnode)
+		if (!Array.isArray(output)) output = [output]
+		vnode.instance = Vnode.normalize(output)
 		sentinel.$$reentrantLock$$ = null
 	}
 	function createComponent(parent, vnode, hooks, ns, nextSibling) {
 		initComponent(vnode, hooks)
 		if (vnode.instance != null) {
-			if (vnode.instance === vnode) throw Error("A view cannot return the vnode it received as arguments")
+			if (vnode.instance[0] === vnode) throw Error("A view cannot return the vnode it received as arguments")
```

Comment:
`vnode.normalize()` turns the array into a fragment, it should be `vnode.instance.children[0]`.

Note that this is the kind of change that will most likely affect perf negatively (adding one `updateNodes()` per component to the call stack), even if we define `vnode.instance = Vnode.normalize(output).children` and change downstream methods to deal with it.